torch/lib/c10d/reducer.cpp

```diff
@@ -1444,6 +1517,45 @@ void Reducer::ensure_prior_reduction_finished() {
       kBaseErrorMsg += kOutputsNotUsedInLossErrorMsg;
       kBaseErrorMsg += kDDPBugErrorMsg;
     }
+
+    const std::string unmarked_param_indices_info = c10::str(
+        "\n",
+        "Parameter indices which did not receive grad for rank ",
+        process_group_->getRank(),
```

Comment:
it is nice to have rank info in the error message here. I also found it is useful to add 'num_iterations_' info for debugging. so it would be nice to add 'num_iterations_' and 'getRank()' to all DDP related error messages in the follow up diff as well.